src/NATS.Client.Core/Internal/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -38,16 +72,6 @@ public static SslClientAuthenticationOptions LoadCaCertsFromPem(this SslClientAu
 
     public static SslClientAuthenticationOptions LoadClientCertFromX509(this SslClientAuthenticationOptions options, X509Certificate2 leafCert, X509Certificate2Collection? intermediateCerts = null, bool offline = false, SslCertificateTrust? trust = null)
     {
-        // On Windows, ephemeral keys/certificates do not work with schannel. e.g. unless stored in certificate store.
-        // https://github.com/dotnet/runtime/issues/66283#issuecomment-1061014225
-        // https://github.com/dotnet/runtime/blob/380a4723ea98067c28d54f30e1a652483a6a257a/src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs#L192-L197
-        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-        {
-            var ephemeral = leafCert;
-            leafCert = new X509Certificate2(leafCert.Export(X509ContentType.Pfx));
-            ephemeral.Dispose();
-        }
```

Comment:
same comment. do we think application code calling this extension should make their own choice about the ephemeral thing?

Reply:
The extensions are internal currently. I was thinking if there is interest we could make them public eventually